### PR TITLE
Update documentation. Move install.sh to run-puppet

### PR DESCRIPTION
**Pull request by e-gris**
I made some documentation and installation changes.

I updated the README with a bunch of hints that I needed to get this running
as a Puppet novice.

I moved the install.sh script to run-puppet and filled out some if the
insides, hopefully without disturbing the original usage.

(Executables shouldn't have an '.sh' suffix and there is already a common command called 'install')

LMK if anything here is objectionable; it has been working well for me.

### CODE REVIEW

File: run-puppet [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  exit 1
+fi
+
+if ! command_exists /opt/puppetlabs/puppet/bin/puppet; then
+  install_puppet
+fi
+
+if [ $r10k_fetch = true ]; then
+  if [ -z "$control_repo" ]; then
+    echo "$script: ERROR control_repo is not set. Use '-c' or the CONTROL_REPO envionment variable." 1>&2
```
Comment (demophoon) [inline run-puppet]: environment

File: run-puppet [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+if [ $UID -ne 0 ]; then
+  echo "$script: ERROR This script must run as root to work properly. Use 'sudo'" 1>&2
+  exit 1
+fi
+
+if ! command_exists /opt/puppetlabs/puppet/bin/puppet; then
+  install_puppet
+fi
+
+if [ $r10k_fetch = true ]; then
```
Comment (demophoon) [inline run-puppet]: `$r10k_fetch` should really be named something like `$r10k_initialize` or something similar.

We are technically using r10k to fetch updated version of the modules every time we call `run_puppet` whereas here we are bootstrapping that environment.

File: run-puppet [hunk 3]
```diff
@@ -0,0 +1,100 @@
+#!/bin/bash -eu
+
+script=$(basename $0)
```
Comment (demophoon) [inline run-puppet]: To make shellcheck happy we should add quotes to variables.
```sh
script=$(basename "$0")
```

File: run-puppet [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+function install_puppet() {
+  if command_exists dpkg; then
+    # Currently only supports ubuntu
+    source /etc/os-release
+    codename=$(lsb_release --codename | cut --fields=2)
+    release_file="puppet-release-$codename.deb"
+    wget https://apt.puppetlabs.com/release_file
+    dpkg --install $release_file
+    rm $release_file
```
Comment (demophoon) [inline run-puppet]: One more shellcheck quoting error
```sh
dpkg --install "$release_file"
rm "$release_file"
```

**Comment by e-gris** [comment]
Would it be worthwhile to put shellcheck in Github to do PR checks?

**Comment by demophoon** [comment]
That is a great idea! I'll add that in just after merging this PR in.

**Pull request merged**